packages/next/src/next-devtools/dev-overlay/components/overview/segment-explorer.tsx

```diff
@@ -59,6 +61,12 @@ function PageSegmentTreeLayerPresentation({
   node: SegmentTrieNode
   level: number
 }) {
+  const [shadowRoot] = useState<ShadowRoot | null>(() => {
+    const portal = document.querySelector('nextjs-portal')
+    if (!portal) return null
+    return portal.shadowRoot as ShadowRoot
+  })
+  const shadowRootRef = useRef<ShadowRoot>(shadowRoot)
```

Comment:
I'm wondering why we need both state and a ref here, especially since the state doesn't need to be updated?

Perhaps should be `useRef` and then we set the ref in an effect?

Reply:
useMemo might [discard the cache and re-caculate it again](https://react.dev/reference/react/useMemo#caveats), useState functional initializer will only do once. I don't feel it's a necessary change